Yoast/wordpress-seo
# Introduce surfaces

## Summary

Introduces new classes to handle the external interface of our plugin from a code perspective.

This PR can be summarized in the following changelog entry:

* Introduces a new function to get the title of the current page: `YoastSEO()->current_page->get_title()`
* Introduces a new function to get the meta description of the current page: `YoastSEO()->current_page->get_description()`
* Introduces a new function to get the canonical of the current page: `YoastSEO()->current_page->get_canonical()`
* Introduces a new function to get the robots values of the current page: `YoastSEO()->current_page->get_robots()`
* Introduces a new function to get a class instance for the purpose of unhooking things: `YoastSEO()->classes->get( Front_End_Integration::class )`

## Relevant technical choices:

* Surface was chosen as the concept for these classes as these are intended to be the outer shell of our plugins that other plugins, themes and custom installations interact with.
* The current page variables currently aren't output through the filters that are used in the presenters. This may be desirable.

## Test instructions
This PR can be tested by following these steps:

* Add some code in your theme that outputs all the above described `current_page` functions. They should match your meta tags.
* Use  the `classes->get` function to get the instance of a class and unhook something ( for example the front end integration ). That hook should no longer be run.

## UI changes

## Documentation

## Quality assurance



## To do:
- [x] Review this PR
- [x] Replace the calls to `yoastseo` with calls to `YoastSEO` in Video, News and Woo in the feature branch, and check if the interface introduced in this PR can be used. Also check other plugins (AMP, ACF, clicky, search index purge, slideshare etc.), and create an indexable feature branch there to fix this.
    - Tip: look for the PRs introducing `yoastseo` to find where to fix it.
